unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -86,6 +86,31 @@ public Interval(int months, long microseconds) {
     this.microseconds = microseconds;
   }
 
+  public Interval doOp(Interval that, String op) {
```

Comment:
i don't think you want this -- it is super slow. Just remove this function, and define codegen in Add / Subtract, rather than relying on what BinaryArithmetic provides.